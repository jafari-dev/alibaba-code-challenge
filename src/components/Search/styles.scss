.search {
  outline: none;
  background-color: #ffffff;
  border-radius: 4px;
  box-shadow: 0 0 4px #cccccc;
  width: 300px;
  display: flex;
  flex-direction: row;
  align-items: center;

  input {
    outline: none;
    padding: 12px 8px;
    border: none;
    width: 272px;
    font-size: 16px;
    font-weight: 600;
    font-family: "Nunito Sans", sans-serif;

    &::placeholder {
      color: #aaaaaa;
    }
  }

  svg {
    margin-left: 8px;
    width: 24px;
    max-width: 24px;
    height: 24px;
  }
}