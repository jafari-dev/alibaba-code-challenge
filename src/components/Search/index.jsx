import "./styles.scss";
import { memo, useCallback } from "react";
import { Search as SearchIcon } from "../../assets";

function Search({ searchValue, onChange }) {
  const handleChangeValue = useCallback(
    (event) => {
      const value = event.currentTarget.value;
      onChange(value);
    },
    [onChange]
  );

  return (
    <div className="search">
      <SearchIcon />
      <input
        placeholder="Search for a country ..."
        value={searchValue}
        onChange={handleChangeValue}
      />
    </div>
  );
}

export default memo(Search);
