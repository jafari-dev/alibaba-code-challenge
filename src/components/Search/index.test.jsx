/* eslint-disable testing-library/prefer-screen-queries */
import { fireEvent, render } from "@testing-library/react";
import { faker } from "@faker-js/faker";
import Search from ".";

beforeAll(() => {
  faker.seed(1);
});

describe("Search component", () => {
  test("matches the snapshots.", () => {
    const { container } = render(
      <Search searchValue={faker.random.words()} onChange={jest.fn()} />
    );

    expect(container).toMatchSnapshot();
  });

  test("sets the provided `searchValue` as the input's value attribure.", () => {
    const searchValue = faker.random.words();

    const { getByRole } = render(
      <Search searchValue={searchValue} onChange={jest.fn()} />
    );

    expect(getByRole("textbox")).toHaveValue(searchValue);
  });

  test("calls the provided `onChange` by entered value in the input.", () => {
    const onChange = jest.fn();

    const { getByRole } = render(
      <Search searchValue={faker.random.words()} onChange={onChange} />
    );

    const enteredValue = faker.random.words();

    fireEvent.change(getByRole("textbox"), { target: { value: enteredValue } });
    expect(onChange).toBeCalledWith(enteredValue);
  });
});
