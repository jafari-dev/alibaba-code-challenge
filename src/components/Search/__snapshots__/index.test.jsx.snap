// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component matches the snapshots. 1`] = `
<div>
  <div
    class="search"
  >
    <svg>
      search.svg
    </svg>
    <input
      placeholder="Search for a country ..."
      value="remove yuck"
    />
  </div>
</div>
`;
